packages/compass-saved-aggregations-queries/src/stores/aggregations-queries-items.ts

```diff
@@ -1,5 +1,102 @@
-function reducer(state = {}): typeof state {
+import { remote } from 'electron';
+import { join } from 'path';
+import fs from 'fs';
+import { AnyAction } from 'redux';
+import { FavoriteQueryCollection } from '@mongodb-js/compass-query-history';
+
+import { actions } from '../actions/aggregations-queries-actions';
+interface Item {
+  id: string;
+  name: string;
+  namespace: string;
+  lastModified: number;
+}
+
+export interface Aggregation extends Item {
+  pipeline: unknown[];
+}
+
+export interface Query extends Item {
+  collation: Record<string, unknown>;
+  filter: Record<string, unknown>;
+  limit: number;
+  project: Record<string, unknown>;
+  skip: number;
+  sort: Record<string, number>;
+}
+
+export type State = {
+  loading: boolean;
+  items: Array<Aggregation | Query>;
+};
+
+const INITIAL_STATE: State = {
+  loading: true,
+  items: [],
+};
+
+function reducer(state = INITIAL_STATE, action: AnyAction): State {
+  if (action.type === actions.FETCH_DATA) {
+    return {
+      items: [...getAggregations(), ...getQueries()],
+      loading: false,
+    };
+  }
   return state;
 }
 
+const getAggregations = () => {
+  const dir = join(remote.app.getPath('userData'), 'SavedPipelines');
+  const aggregations = fs
+    .readdirSync(dir)
```

Comment:
This is probably not very great, we shouldn't use the sync API, but moreover, I'd recommend just extracting the function that is used in `compass-aggregations` (https://github.com/mongodb-js/compass/blob/06fd6c3374aea603d5694bc0ab8b0a0260d4eafd/packages/compass-aggregations/src/modules/saved-pipeline.js#L85), if you want we can try to do that together.